tests/other_errors/import_cycle.jou

```diff
@@ -6,17 +6,7 @@ def a(x: int) -> None:
     if x > 0:
         b(x-1)
 
-# Output: a 10
-# Output: b 9
-# Output: a 8
-# Output: b 7
-# Output: a 6
-# Output: b 5
-# Output: a 4
-# Output: b 3
-# Output: a 2
-# Output: b 1
-# Output: a 0
+# Output: compiler error in file "tests/other_errors/import_cycle.jou": imported file should not have `main` function
```

Comment:
This is valid code. It shouldn't fail to compile. It is fine to have a `main` function in an imported file if that is the file name you pass to the Jou compiler.

Maybe instead of this:

```
for each imported file:
    if file has a `main` function:
        error
```

the logic should be more like this:

```
if file is the file given to Jou compiler as argument:
    if no main function:
        error
else:
    if it has a main function:
        error
```

Reply:
> This is valid code. It shouldn't fail to compile. It is fine to have a `main` function in an imported file if that is the file you give to the Jou compiler when you compile your program.
> 
> Maybe instead of this:
> 
> ```
> for each imported file:
>     if file has a `main` function:
>         error
> ```
> 
> the logic should be more like this:
> 
> ```
> if file is the file given to Jou compiler as argument:
>     if no main function:
>         error
> else:
>     if it has a main function:
>         error
> ```

So it may cause:
```
compiler error in file "tests/other_errors/import_file_with_main.jou", line 4: a function named 'main' already exists
```

Reply:
I'm not completely sure how this should work. I'll pour more time into it during the next few days.